Python/ceval.c

```diff
@@ -6935,6 +6935,15 @@ PyEval_SetProfile(Py_tracefunc func, PyObject *arg)
     }
 }
 
+void
+PyEval_SetThreadProfile(PyThreadState *tstate, Py_tracefunc func, PyObject *arg)
+{
+    if (_PyEval_SetProfile(tstate, func, arg) < 0) {
+        /* Log _PySys_Audit() error */
+        _PyErr_WriteUnraisableMsg("in PyEval_SetProfile", NULL);
+    }
+}
```

Comment:
Would it be possible to add a simple test to _testcapi.test_tstate_capi()? Simple: just call the function.

If you want to write a more elaborated test, maybe add test to _testinternalcapi to check c_profilefunc, c_profileobj and tracing members of PyThreadState.

The test should then call PyEval_SetThreadProfile(tstate, NULL, NULL) to clear the function. I don't think that it's worth it to save/restore the old value. It shouldn't be used by tests like test_capi.